contrib/platform/src/com/sun/jna/platform/win32/WinDef.java

```diff
@@ -1774,4 +1774,54 @@ public HGLRCByReference(HGLRC h) {
             super(h);
         }
     }
+
+    /**
+     * COORD structure
+     */
+    @FieldOrder({ "X", "Y" })
+    public static class COORD extends Structure {
+
+        public short X;
+        public short Y;
+
+        @Override
+        public String toString() {
+            return String.format("COORD(%s,%s)", X, Y);
+        }
+    }
+
+    /**
+     * SMALL_RECT structure
+     */
+    @FieldOrder({ "Left", "Top", "Right", "Bottom" })
+    public static class SMALL_RECT extends Structure {
+
+        public short Left;
+        public short Top;
+        public short Right;
+        public short Bottom;
+
+        @Override
+        public String toString() {
+            return String.format("SMALL_RECT(%s,%s)(%s,%s)", Left, Top, Right, Bottom);
+        }
+    }
+
+    /**
+     * CONSOLE_SCREEN_BUFFER_INFO structure
+     */
+    @FieldOrder({ "dwSize", "dwCursorPosition", "wAttributes", "srWindow", "dwMaximumWindowSize" })
+    public static class CONSOLE_SCREEN_BUFFER_INFO extends Structure {
+
+        public COORD dwSize;
+        public COORD dwCursorPosition;
+        public WORD wAttributes;
```

Comment:
`COORD` and `SMALL_RECT` both map to java shorts. It would be an option to also map wAttributes  to java short. If I read the documentation correctly it is a set of shorts ORed together, so the sign is not relevant.

Reply:
I thought similar when I looked at it, but decided the plain structure interpretation (`dwSize.X` and `dwSize.Y`) would be preferred over an `int` with bitshifting/bitmasking getters to access the underlying values.  About the only thing I might change in this mapping is that `wAttributes` as a `short` would more directly map to the WINAPI `enum` values it represents.

Reply:
Er... that's what you said. I misread it the first time.